open_rarity/resolver/testset_resolver.py

```diff
@@ -205,90 +165,82 @@ def resolve_collection_data(
-            )
-            tokens_with_rarity: list[
-                TokenWithRarityData
-            ] = get_tokens_with_rarity(
-                collection_with_metadata=collection_with_metadata,
-                resolve_remote_rarity=resolve_remote_rarity,
-                max_tokens_to_calculate=max_tokens_to_calculate,
+    if not golden_collections:
+        raise ValueError("Can't resolve golden collections data file.")
+
+    data = json.load(io.BytesIO(golden_collections))
+    for collection_def in data:
+        opensea_slug = collection_def["collection_slug"]
+        # Fetch collection metadata and tokens that belong to this collection
+        # from opensea and other external api's.
+        collection_with_metadata = get_collection_with_metadata_from_opensea(
+            opensea_collection_slug=opensea_slug
+        )
+        tokens_with_rarity: list[TokenWithRarityData] = get_tokens_with_rarity(
+            collection_with_metadata=collection_with_metadata,
+            resolve_remote_rarity=resolve_remote_rarity,
+            max_tokens_to_calculate=max_tokens_to_calculate,
+        )
+        collection = collection_with_metadata.collection
+        collection.tokens = [tr.token for tr in tokens_with_rarity]
+        if max_tokens_to_calculate is None:
+            assert collection.token_total_supply == len(tokens_with_rarity)
+        else:
+            assert max_tokens_to_calculate == len(tokens_with_rarity)
+
+        # Calculate and append open rarity scores
+        if max_tokens_to_calculate is None:
+            open_rarity_scores = resolve_open_rarity_score(
+                collection, collection.tokens, normalized=True
             )
-            collection = collection_with_metadata.collection
-            collection.tokens = [tr.token for tr in tokens_with_rarity]
-            if max_tokens_to_calculate is None:
-                assert collection.token_total_supply == len(tokens_with_rarity)
-            else:
-                assert max_tokens_to_calculate == len(tokens_with_rarity)
-
-            # Calculate and append open rarity scores
-            if max_tokens_to_calculate is None:
-                open_rarity_scores = resolve_open_rarity_score(
-                    collection, collection.tokens, normalized=True
-                )
-                augment_with_open_rarity_scores(
-                    tokens_with_rarity=tokens_with_rarity,
-                    scores=open_rarity_scores,
-                )
-
-            serialize_to_csv(
-                collection_with_metadata=collection_with_metadata,
+            augment_with_open_rarity_scores(
                 tokens_with_rarity=tokens_with_rarity,
+                scores=open_rarity_scores,
             )
 
-    else:
-        raise Exception("Can't resolve golden collections data file.")
+        serialize_to_csv(
+            collection_with_metadata=collection_with_metadata,
+            tokens_with_rarity=tokens_with_rarity,
+        )
```

Comment:
We should allow for other file extensions. People are probably going to generally prefer json since we are in the web space typically.

Maybe make a generic serializer that handles file type?

Reply:
let's do this in follow-up PR if needed - this was just what we had before